homeassistant/components/workday/binary_sensor.py

```diff
@@ -113,17 +113,24 @@ async def async_setup_entry(
     """Set up the Workday sensor."""
     add_holidays: list[DateLike] = entry.options[CONF_ADD_HOLIDAYS]
     remove_holidays: list[str] = entry.options[CONF_REMOVE_HOLIDAYS]
-    country: str = entry.options[CONF_COUNTRY]
+    country: str | None = entry.options.get(CONF_COUNTRY)
     days_offset: int = int(entry.options[CONF_OFFSET])
     excludes: list[str] = entry.options[CONF_EXCLUDES]
     province: str | None = entry.options.get(CONF_PROVINCE)
     sensor_name: str = entry.options[CONF_NAME]
     workdays: list[str] = entry.options[CONF_WORKDAYS]
 
     year: int = (dt.now() + timedelta(days=days_offset)).year
-    obj_holidays: HolidayBase = getattr(holidays, country)(years=year)
+
+    if country:
+        obj_holidays: HolidayBase = getattr(holidays, country)(years=year)
+    else:
+        obj_holidays = HolidayBase(years=year)
 
     if province:
+        if not country:
+            LOGGER.error("Cannot use a subdivision/province without a country")
```

Comment:
This can't happen as we (should) skip province in the schema if country is `None`

Reply:
Agreed, but there must be some sort of hint to mypy that `country` is non-`None` here for `getattr(holidays, country)` to type check. What is the preferred style in Home Assistant? I will add `if typing.TYPE_CHECKING: assert country is not None` to fix for now, but please let me know which is best.

Reply:
I think you should turn it around `if country and province:` perhaps. Didn't look to careful though